edgelet/iotedge/src/check/checks/host_connect_dps_endpoint.rs

```diff
@@ -102,3 +128,37 @@ pub fn resolve_and_tls_handshake(
 
     Ok(())
 }
+
+pub fn resolve_and_tls_handshake_proxy(
+    tls_hostname: String,
+    port: Option<u16>,
+    proxy: String,
+) -> impl Future<Item = (), Error = Error> {
+    futures::future::ok(())
+        .and_then({
+            move |_| -> Result<_, Error> {
+                let proxy_uri = Uri::from_str(&proxy)
+                    .with_context(|_| format!("Could not make proxi uri from {}", proxy))?;
+                let proxy_obj = Proxy::new(Intercept::All, proxy_uri);
+                let http_connector = HttpConnector::new(1);
+                let proxy_connector = ProxyConnector::from_proxy(http_connector, proxy_obj)
```

Comment:
For TLS proxies `HttpConnector` won't be sufficient. You'll need an `HttpsConnector`.

You don't need to check the scheme. Just always use an `HttpsConnector`. It can handle `http://` too.

Reply:
Done. Had to add hyper_tls to the project